tests/scene/test_primitives.h

```diff
@@ -845,6 +846,236 @@ TEST_CASE("[SceneTree][Primitive][Text] Text Primitive") {
+		mesh->set_subdivide_width(i & 1);
+		mesh->set_subdivide_height(i & 2);
+		mesh->set_subdivide_depth(i & 4);
+		CHECK(mesh->is_surface_quads(0));
+	}
+
+	for (int i = 0; i < 4; i++) {
+		Ref<PlaneMesh> mesh = memnew(PlaneMesh);
+		mesh->set_subdivide_width(i & 1);
+		mesh->set_subdivide_depth(i & 2);
+		CHECK(mesh->is_surface_quads(0));
+	}
+
+	{
+		Ref<TorusMesh> mesh = memnew(TorusMesh);
+		CHECK(mesh->is_surface_quads(0));
+		mesh->set_flip_faces(true);
+		CHECK(mesh->is_surface_quads(0));
+	}
+	{
+		Ref<RibbonTrailMesh> mesh = memnew(RibbonTrailMesh);
+		CHECK(mesh->is_surface_quads(0));
+		mesh->set_flip_faces(true);
+		CHECK(mesh->is_surface_quads(0));
+	}
+
+	{
+		Ref<PrismMesh> mesh = memnew(PrismMesh);
+		CHECK_FALSE(mesh->is_surface_quads(0));
+	}
+
+	// The triangles in these meshes are actually implemented as degenerate quads for simplicity.
+	{
+		Ref<CapsuleMesh> mesh = memnew(CapsuleMesh);
+		CHECK(mesh->is_surface_quads(0));
+	}
+	{
+		Ref<SphereMesh> mesh = memnew(SphereMesh);
+		CHECK(mesh->is_surface_quads(0));
+	}
+
+	// Fully quads when uncapped.
+	{
+		Ref<CylinderMesh> mesh = memnew(CylinderMesh);
+		mesh->set_radial_segments(11);
+		CHECK_FALSE(mesh->is_surface_quads(0));
+		mesh->set_cap_top(false);
+		mesh->set_cap_bottom(false);
+		CHECK(mesh->is_surface_quads(0));
+	}
+	{
+		Ref<TubeTrailMesh> mesh = memnew(TubeTrailMesh);
+		mesh->set_radial_steps(11);
+		CHECK_FALSE(mesh->is_surface_quads(0));
+		mesh->set_cap_top(false);
+		mesh->set_cap_bottom(false);
+		CHECK(mesh->is_surface_quads(0));
+	}
+
+	// All nine ways that vertices in triangle pairs can be ordered.
+	Vector<int> indices = {
+		111, 222, 333, 444, 111, 333,
+		111, 222, 333, 111, 333, 444,
+		111, 222, 333, 333, 444, 111,
+
+		222, 333, 111, 444, 111, 333,
+		222, 333, 111, 111, 333, 444,
+		222, 333, 111, 333, 444, 111,
+
+		333, 111, 222, 444, 111, 333,
+		333, 111, 222, 111, 333, 444,
+		333, 111, 222, 333, 444, 111
+	};
+	// The canonical order that the above index array should be sorted into.
+	Vector<int> indices_quads_expect = {
+		111, 222, 333, 444, 111, 333,
+		111, 222, 333, 444, 111, 333,
+		111, 222, 333, 444, 111, 333,
+		111, 222, 333, 444, 111, 333,
+		111, 222, 333, 444, 111, 333,
+		111, 222, 333, 444, 111, 333,
+		111, 222, 333, 444, 111, 333,
+		111, 222, 333, 444, 111, 333,
+		111, 222, 333, 444, 111, 333
+	};
+
+	CHECK_FALSE(indices == indices_quads_expect);
+
+	CHECK_FALSE(SurfaceTool::is_indices_quads(indices.ptr(), indices.size()));
+	SurfaceTool::set_indices_quads(indices.ptrw(), indices.size());
+	CHECK(SurfaceTool::is_indices_quads(indices.ptr(), indices.size()));
+
+	CHECK(indices == indices_quads_expect);
+}
+
+static void dump_as_obj(String p_path, const Array &p_array) {
+	IF_DEV_VAR("DUMP_TEST_MESHES", {
+		const Vector<int> &src_indices = p_array[Mesh::ARRAY_INDEX];
+		const Vector<Vector3> &src_vertices = p_array[Mesh::ARRAY_VERTEX];
+
+		Ref<FileAccess> file = FileAccess::open("/tmp/" + p_path, FileAccess::WRITE);
```

Comment:
This test will fail on Windows, as there's no `/tmp` there. Use an `#ifdef` that uses `OS::get_singleton()->get_environment("TEMP").path_join(...)` when on Windows.